software/py/blood_graph_key.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import argparse
+from blood_graph import BloodGraph
+from PIL import Image, ImageDraw, ImageFont
+from itertools import chain
+
+WIDTH  = 256
+HEIGHT = 256
+DEFAULT_MODE = "detailed"
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Depicts a visual key for bloodgraph")
+    parser.add_argument("--mode", default=DEFAULT_MODE, type=str,
+                        help="Type of bloodgraph - abstract / detailed")
+    args = parser.parse_args()
+
+    if (not (args.mode == "abstract" or args.mode == "detailed")):
+        parser.error("Invalid mode, can be one of abstract / detailed.")
+        sys.exit()
+
+    return args
+
+
+
+if __name__ == "__main__":
+    args = parse_args()
+
+    bg = BloodGraph(0,1,1, args.mode)
```

Comment:
Does BloodGraph automatically generate an output image?

Reply:
No

Reply:
You need to call generate() to do that.

Reply:
Still want to know why creating a BloodGraph object is necessary

Reply:
it was to harvest it for its internal members; moot now though as I've unified the scripts.